lib/Target/SubstraitPB/Export.cpp

```diff
@@ -96,6 +155,20 @@ FailureOr<std::unique_ptr<proto::Type>> exportType(Location loc,
   return emitError(loc) << "could not export unsupported type " << mlirType;
 }
 
+/// We just forward to the overload for `RelOpInterface`, which will have to
+/// export this op. We can't (easily) do it here because the emit op is
+/// represented as part of the `RelCommon` message of one of the cases of the
+/// `Rel` message but there is no generic way to access the `common` field of
+/// the various cases.
+FailureOr<std::unique_ptr<Rel>> exportOperation(EmitOp op) {
```

Comment:
I am not 100% happy with the whole structure. It is somewhat dangerous to rely on implementations doing a particular thing without being able to enforce it.

The problem is, I think, that I alluded to here: there is no proper way to the access the `common` field of different message types. Maybe it's worth to consider the following alternative: implement a `accessCommonField` function with a big switch that accesses that field for any of the cases we support. This is ugly but in concentrates the ugliness in one place. Then `exportOperation(EmitOp)` could first export the op that produces its input and then *modify* the `common` field of the resulting message of that export. I suppose something similar could work for the import.